### PR TITLE
Typescript ambient types added

### PR DESCRIPTION
**Pull request by gorillatron**
I added some ambient types for Typescript. All tests passed. No interference with source code itself.

Love this module! 🍻 ❤️ 😄

### CODE REVIEW

File: typings.d.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  export type IResource = IResourceGetter | IResourceProp | string
+
+  export interface IRoleProp {
+    role_id:string;
+  }
+
+  export interface IRoleGetter {
+    getRoleId():string;
```
Comment (spruce-bruce) [inline typings.d.ts]: I don't know anything about type declarations, but this can also return an array.

File: typings.d.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  export class Acl {
+
+    addRole(role:string, parent?:string):void
+
+    addResource(resource:string, parent?:string):void
+
+    allow<Ro extends IRole, Re extends IResource> (role?:string, resource?:string, actions?:string[], assertion?:boolean|IAssertion<Ro, Re>):void
+
+    deny<Ro extends IRole, Re extends IResource> (role?:string, resource?:string, actions?:string[], assertion?:boolean|IAssertion<Ro, Re>):void
```
Comment (spruce-bruce) [inline typings.d.ts]: role here can also be an array

**Comment by gorillatron** [comment]
Thank you! I will submit a commit later today with your corrections 🙏😌

**Comment by djvirgen** [comment]
This looks good, thank you so much for doing this! I'm familiar with TypeScript but haven't used it much myself. I'm looking forward to seeing how this can help with your fellow TypeScript coders!

I'll merge this and publish a new version now. If you run into any issues please file a bug report here. Thank you!

**Comment by djvirgen** [comment]
virgen-acl@0.0.22 has been published to npm. Thanks again!

**Pull request merged**